index.js

```diff
@@ -76,5 +76,12 @@ export default async function pRetry(input, options) {
 				}
 			}
 		});
+
+		if (options.signal && !options.signal.aborted) {
+			options.signal.addEventListener('abort', () => {
+				operation.stop();
+				reject(new Error('Operation aborted by AbortSignal.'));
```

Comment:
We could use the `AbortSignal.reason` property instead of a generic message (it's given the value passed to `AbortController.abort()` as argument).

It's only supported in latest Firefox and Node >=16.14.0 though.

Reply:
> It's only supported in latest Firefox and Node >=16.14.0 though.

That's fine. I prefer using that.